sale_triple_discount/models/sale_order.py

```diff
@@ -18,27 +18,22 @@ def _amount_all(self):
         Compute the total amounts of the SO.
         """
         for order in self:
-            if not order.company_id.tax_calculation_rounding_method == (
-                    'round_globally'):
-                return super(SaleOrder, self)._amount_all()
-            amount_untaxed = amount_tax = 0.0
+            prev_price_unit = {}
             for line in order.order_line:
-                amount_untaxed += line.price_subtotal
-                taxes = line.tax_id.compute_all(
-                    line.price_reduce,
-                    line.order_id.currency_id,
-                    line.product_uom_qty,
-                    product=line.product_id,
-                    partner=order.partner_shipping_id,
-                )
-                amount_tax += sum(
-                    t.get('amount', 0.0) for t in taxes.get('taxes', []))
-            order.update({
-                'amount_untaxed': order.pricelist_id.currency_id.round(
-                    amount_untaxed),
-                'amount_tax': order.pricelist_id.currency_id.round(amount_tax),
-                'amount_total': amount_untaxed + amount_tax,
-            })
+                prev_price_unit['line.id'] = line.price_unit
+                prev_price_subtotal = line.price_subtotal
```

Comment:
You're still not following the same scheme than in the `_compute_amount` of `sale.order.line` model.

Reply:
I'm tried to follow same scheme than in the _compute_amount of sale.order.line mode, but it doesn't work, I think because updating sale order line values trigers again _amount_all function in sale.order model ...

For example, this code is wrong:

    @api.depends('order_line.price_total')
    def _amount_all(self):
        for order in self:
            prev_price_unit = {}
            prev_discount = prev_discount2 = prev_discount3 ={}
            for line in order.order_line:
                prev_discount['line.id'] = line.discount
                prev_discount2['line.id'] = line.discount2
                prev_discount3['line.id'] = line.discount3
                prev_price_unit['line.id'] = line.price_unit
                price_unit = line.price_unit * (
                    1 - (line.discount or 0.0) / 100.0) * (
                                 1 - (line.discount2 or 0.0) / 100.0) * (
                                 1 - (line.discount3 or 0.0) / 100.0)
                line.update({
                    'price_unit': price_unit,
                    'discount': 0.0,
                    'discount2': 0.0,
                    'discount3': 0.0,
                })
            super(SaleOrder, self)._amount_all()
            for line in order.order_line:
                line.update({
                    'price_unit': prev_price_unit['line.id'],
                    'discount': prev_discount['line.id'],
                    'discount2': prev_discount2['line.id'],
                    'discount3': prev_discount3['line.id'],
                })

The only way I find is rewriting the function in order to calculate the new price inside.